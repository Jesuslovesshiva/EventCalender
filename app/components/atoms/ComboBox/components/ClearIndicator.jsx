import classNames from 'classnames'
import PropTypes from 'prop-types'
import {components} from 'react-select'

import CloseIcon from '@/components/atoms/Icon/Close'

/* eslint-disable react/jsx-props-no-spreading */
const ClearIndicator = ({className, ...props}) => (
  <components.ClearIndicator
    className={classNames(
      'text-2xl !py-1 text-gray-medium hover:!text-gray-dark/50 hover:cursor-pointer !px-0',
      className
    )}
    {...props}
  >
    <CloseIcon />
  </components.ClearIndicator>
)
/* eslint-enable */

ClearIndicator.propTypes = {
  className: PropTypes.string
}

ClearIndicator.defaultProps = {
  className: null
}

export default ClearIndicator
