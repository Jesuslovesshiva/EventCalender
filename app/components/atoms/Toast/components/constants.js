const TOAST_VARIANTS = {
  ERROR: 'error',
  SUCCESS: 'success'
}

const AVAILABLE_TOAST_VARIANTS = Object.values(TOAST_VARIANTS)

const TOAST_TYPES = {
  ACTION: 'action',
  BASIC: 'basic',
  CLOSE_ACTION: 'close',
  DESCRIPTION: 'description'
}

const AVAILABLE_TOAST_TYPES = Object.values(TOAST_TYPES)

export {
  AVAILABLE_TOAST_TYPES,
  AVAILABLE_TOAST_VARIANTS,
  TOAST_TYPES,
  TOAST_VARIANTS
}
