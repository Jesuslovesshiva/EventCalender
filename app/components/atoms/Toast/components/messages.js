import {defineMessages} from 'react-intl'

const scope = 'atoms.toast'

export default defineMessages({
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Info'
  },
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Error or Warning'
  },
  alert: {
    id: `${scope}.alert`,
    defaultMessage: 'alert'
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'Action'
  }
})
