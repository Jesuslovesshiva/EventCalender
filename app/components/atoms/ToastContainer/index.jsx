import classNames from 'classnames'
import {ToastContainer as ToastContainerComponent} from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'

const ToastContainer = () => (
  <ToastContainerComponent
    autoClose={4000}
    closeButton={false}
    closeOnClick={false}
    draggable
    hideProgressBar
    newestOnTop
    pauseOnFocusLoss={false}
    pauseOnHover={false}
    position="bottom-center"
    theme="colored"
    toastClassName={() =>
      classNames(
        'bg-gradient-to-l bg-gradient-primary text-xl',
        'min-h-[41px] h-fit min-w-[437px] w-fit border border-white rounded',
        'flex justify-center items-center px-2.5 my-1'
      )
    }
    bodyStyle={{alignItems: 'start'}}
  />
)

export default ToastContainer
