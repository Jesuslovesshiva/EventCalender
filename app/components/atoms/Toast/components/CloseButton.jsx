import noop from 'lodash/noop'
import PropTypes from 'prop-types'

import Button, {BUTTON_TYPES} from '@/components/atoms/Button'
import CloseIcon from '@/components/atoms/Icon/Close'
import messages from '@/components/atoms/Toast/components/messages'
import useTranslation from '@/hooks/use-translation'
import customPropTypes from '@/utils/prop-types'

import {AVAILABLE_TOAST_TYPES, TOAST_TYPES} from './constants'

const CloseButton = ({onClick, onClose, text, type}) => {
  if (type === TOAST_TYPES.ACTION) {
    const message = useTranslation(text)

    return (
      <Button
        className="text-white hover:!bg-transparent my-[11px]"
        onClick={onClick}
        type={BUTTON_TYPES.OUTLINED}
      >
        {message}
      </Button>
    )
  }

  if (type === TOAST_TYPES.CLOSE_ACTION) {
    return (
      <CloseIcon
        className="text-gray-medium hover:text-white cursor-pointer"
        onClick={onClose}
      />
    )
  }

  return null
}

CloseButton.propTypes = {
  onClick: PropTypes.func,
  onClose: PropTypes.func,
  text: customPropTypes.Translatable,
  type: PropTypes.oneOf(AVAILABLE_TOAST_TYPES).isRequired
}

CloseButton.defaultProps = {
  onClick: noop,
  onClose: noop,
  text: messages.action
}

export default CloseButton
