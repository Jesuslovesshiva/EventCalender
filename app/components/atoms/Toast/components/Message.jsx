import classNames from 'classnames'
import PropTypes from 'prop-types'
import {useMemo} from 'react'
import {FormattedMessage} from 'react-intl'

import useTranslation from '@/hooks/use-translation'
import customPropTypes from '@/utils/prop-types'

import {
  AVAILABLE_TOAST_TYPES,
  AVAILABLE_TOAST_VARIANTS,
  TOAST_TYPES
} from './constants'
import messages from './messages'

const Message = ({text, type, variant}) => {
  const isDescription = type === TOAST_TYPES.DESCRIPTION
  const toastMessage = useTranslation(text)
  const message = useMemo(() => messages[variant], [variant])

  return (
    <div
      className={classNames('-ml-1', {
        'flex flex-col pt-1 pb-1.5': isDescription
      })}
    >
      <span
        className={classNames('font-bold inline-flex gap-0.5', {
          'text-base': !isDescription
        })}
      >
        <FormattedMessage {...message} />
        {!isDescription && (
          <>
            {' '}
            <FormattedMessage {...messages.alert} />
          </>
        )}
      </span>
      <span className="text-base">
        {!isDescription && ' - '}
        {toastMessage}
      </span>
    </div>
  )
}

Message.propTypes = {
  text: customPropTypes.Translatable,
  type: PropTypes.oneOf(AVAILABLE_TOAST_TYPES).isRequired,
  variant: PropTypes.oneOf(AVAILABLE_TOAST_VARIANTS).isRequired
}

Message.defaultProps = {
  text: messages.action
}

export default Message
