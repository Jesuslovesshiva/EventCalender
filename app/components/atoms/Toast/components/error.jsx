import classNames from 'classnames'
import {toast} from 'react-toastify'

import WarningAmberIcon from '@/components/atoms/Icon/WarningAmber'

import CloseButton from './CloseButton'
import Message from './Message'
import {TOAST_TYPES, TOAST_VARIANTS} from './constants'

const error = ({
  actionText,
  onActionClick,
  text,
  type = TOAST_TYPES.CLOSE_ACTION
}) => {
  const isDescription = type === TOAST_TYPES.DESCRIPTION

  return toast.error(
    <Message text={text} type={type} variant={TOAST_VARIANTS.ERROR} />,
    {
      autoClose: false,
      closeButton: ({closeToast}) => (
        <CloseButton
          onClick={onActionClick}
          onClose={closeToast}
          text={actionText}
          type={type}
        />
      ),
      hideProgressBar: true,
      icon: () => (
        <div className={classNames({'pt-[3px]': isDescription})}>
          <WarningAmberIcon />
        </div>
      ),
      position: 'bottom-center'
    }
  )
}

export default error
