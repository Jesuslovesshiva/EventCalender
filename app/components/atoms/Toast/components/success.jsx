import classNames from 'classnames'
import {toast} from 'react-toastify'

import InfoIcon from '@/components/atoms/Icon/Info'

import CloseButton from './CloseButton'
import Message from './Message'
import {TOAST_TYPES, TOAST_VARIANTS} from './constants'

const success = ({actionText, onActionClick, text, type}) => {
  const isDescription = type === TOAST_TYPES.DESCRIPTION

  return toast.success(
    <Message text={text} type={type} variant={TOAST_VARIANTS.SUCCESS} />,
    {
      closeButton: ({closeToast}) => (
        <CloseButton
          onClick={onActionClick}
          onClose={closeToast}
          text={actionText}
          type={type}
        />
      ),
      hideProgressBar: true,
      icon: (
        <div className={classNames({'pt-[3px]': isDescription})}>
          <InfoIcon />
        </div>
      ),
      position: 'bottom-center'
    }
  )
}
export default success
