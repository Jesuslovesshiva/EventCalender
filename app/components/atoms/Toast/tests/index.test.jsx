import {fireEvent} from '@testing-library/react'

import ToastContainer from '@/components/atoms/ToastContainer'
import render from '@/utils/test-utils/custom-render'

import {TOAST_TYPES, TOAST_VARIANTS} from '../components/constants'
import toast from '../index'

const defaultProps = {
  text: 'Success toast!',
  type: TOAST_TYPES.BASIC,
  variant: TOAST_VARIANTS.SUCCESS
}

const TestComponent = props => {
  const onNotify = toast(props)

  return <button onClick={onNotify}>toast</button>
}

const renderComponent = (props = {}) =>
  render(
    <>
      <ToastContainer />
      <TestComponent {...defaultProps} {...props} />
    </>
  )

describe('Toast', () => {
  it('should match the snapshot for unknown variant', () => {
    const {container} = renderComponent({type: null, variant: 'unknown'})
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render basic success toast', async () => {
    const {getByText, findByRole} = renderComponent()
    fireEvent.click(getByText('toast'))
    const successToast = await findByRole('alert')
    expect(successToast).toHaveTextContent(defaultProps.text)
  })

  it('should render action error toast', async () => {
    const props = {
      actionText: 'undo',
      onActionClick: jest.fn(),
      text: 'some error',
      type: TOAST_TYPES.ACTION,
      variant: TOAST_VARIANTS.ERROR
    }
    const {getByText, findByRole} = renderComponent(props)
    fireEvent.click(getByText('toast'))
    const errorToast = await findByRole('alert')
    expect(errorToast).toHaveTextContent(props.text)
  })

  it('should render close-action error toast', async () => {
    const props = {
      onActionClick: jest.fn(),
      text: 'some error',
      type: TOAST_TYPES.CLOSE_ACTION,
      variant: TOAST_VARIANTS.ERROR
    }
    const {getByText, findByRole} = renderComponent(props)
    fireEvent.click(getByText('toast'))
    const errorToast = await findByRole('alert')
    expect(errorToast).toHaveTextContent(props.text)
  })

  describe('events', () => {
    it('should call onActionClick function on click', async () => {
      const props = {
        actionText: 'undo',
        onActionClick: jest.fn(),
        type: TOAST_TYPES.ACTION
      }
      const {getByText, findByRole} = renderComponent(props)
      fireEvent.click(getByText('toast'))
      const errorToast = await findByRole('alert')
      const action = errorToast.parentElement.querySelector('button')
      fireEvent.click(action)
      expect(props.onActionClick).toHaveBeenCalled()
    })
  })
})
