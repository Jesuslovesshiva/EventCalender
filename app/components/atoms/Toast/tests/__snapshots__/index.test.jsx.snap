// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should match the snapshot for unknown variant 1`] = `
<div
  class="Toastify"
/>
`;
